dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java

```diff
@@ -242,6 +243,20 @@ private long countAllPublished() {
         return list(this.currentSession().getNamedQuery("io.dockstore.webservice.core.Entry.findLabelByEntryId").setParameter("entryId", entryId));
     }
 
+    public List<String> getToolsDescriptorTypes(long entryId) {
+        return (List<String>)this.currentSession().getNamedQuery("Entry.findToolsDescriptorTypes").setParameter("entryId", entryId)
+                .getSingleResult();
+    }
+
+    public List<String> getWorkflowsDescriptorTypes(long entryId) {
+        List result = this.currentSession().getNamedQuery("Entry.findWorkflowsDescriptorTypes").setParameter("entryId", entryId).getResultList();
+        if (!result.isEmpty()) {
+            return Arrays.asList(this.currentSession().getNamedQuery("Entry.findWorkflowsDescriptorTypes").setParameter("entryId", entryId).getSingleResult().toString());
```

Comment:
It doesn't seem like you should need to execute the query twice like this. Can't you instead use the existing `result` variable?

Reply:
`result` in this case is a list of `enum`, while line 254 is making the `enum` to string and then convert it to a list of string. The reason of me doing this is to make sure an error was not thrown while doing `getSingleResult`(I forgot what it was that was a while ago)

Reply:
Ditto with @coverbeck 's observation. 
All 254 adds is a toString.

Reply:
@denis-yuen might mot be easy to spot the difference, line 252 is `getResultList` as compared to line 254 `getSingleResult`

Reply:
I suggest something like, untested:

```
        List<DescriptorLanguage> result = this.currentSession().getNamedQuery("Entry.findWorkflowsDescriptorTypes").setParameter("entryId", entryId).getResultList();
        return result.stream().map(descriptorLanguage -> descriptorLanguage.toString()).collect(Collectors.toList());
```

1. One query instead of two
2. Returns an empty rather than a null list.